### ISSUE TITLE
Falco upgrade from ebpf to modern-ebpf (v3.5.0) is cause a node damage in AWS-EKS and Azure

### ISSUE THREAD

**Issue by natalia-k** [falcosecurity/falco#2795]
**Describe the bug**
After upgrade from ebpf to modern-ebpf (v3.5.0) we notice a strange behavior in production nodes - first in AWS Graviton nodes and then in Azure
AWS - Nodes (EC2’s) got stuck during their healthcheck in “1/2 checks passed” with the error - “Instance reachability check failed” which cause the nodes to drain all the pods, new nodes from the same type is getting the same error. The instances are not accessible and has no network, once Falco is deleted (runs as daemonset) the node recovered.
nothing relevant was found in logs. ASW support also didn't find anything.
Azure - Node got “node not ready” and got drain, same as in AWS nodes was’t accessible.

During my investigation, I am able to reproduce the issue on eks stg cluster on m6g.xlarge nodes.

**How to reproduce it**

Just install modern-ebpf (v3.5.0) on EKS cluster with Graviton nodes (for example: m6g.xlarge)

**Environment**

- Falco version:
Upgrade: 
Chart: From 2.4.3 to 3.5.0
App: From 0.33.1 to 0.35.1
prob - ebpf to modern-ebpf

- Cloud provider or hardware configuration:
v1.25.12-eks

- Kernel:
5.10.186-179.751.amzn2.aarch64
- Installation method:
helm

**Screenshots**

![image](https://github.com/falcosecurity/falco/assets/4034249/c6d57782-e75f-428a-9df3-fd723952b95e)
<img width="1159" alt="image" src="https://github.com/falcosecurity/falco/assets/4034249/72ead4f8-0dd6-46ac-ae00-350726583a21">

**Comment by leogr** [comment]
> Chart: From 2.4.3 to 3.5.0
> App: From 0.33.1 to 0.35.1
> prob - ebpf to modern-ebpf

Have you tried to use Falco 0.35.1 but keep the (legacy) eBPF probe? :thinking: 
Just to make sure the problem is exclusively linked to the modern probe.

**Comment by leogr** [comment]
cc @Andreagit97

**Comment by Andreagit97** [comment]
ei @natalia-k I'm sorry to hear that :( If I understood correctly you are able to reproduce the issue in a reliable way... So the good news is that we are realising Falco 0.36 in these weeks and we may already solved this issue!
* https://github.com/falcosecurity/libs/pull/1236 
* https://github.com/falcosecurity/libs/pull/1251

Some questions:
- As @leogr correctly pointed out: Have you tried to use Falco 0.35.1 but keep the (legacy) eBPF probe? Just to make sure the problem is exclusively linked to the modern probe.
- Would you be available to test some dev images of Falco 0.36 to check if the issue is still there?
- Can I ask you about the falco config you used in Falco 0.35.1?

**Comment by natalia-k** [comment]
Hi @Andreagit97, Let me first test  Falco 0.35.1 with both (legacy) eBPF & modern-ebpf, I will update. Thanks

**Comment by natalia-k** [comment]
@Andreagit97 
there was no issue with (legacy) eBPF on 0.35.1, I run it during 1h, once I change the config from "kind: ebpf" to "kind: modern-bpf" and redeploy falco, the problem back after few minutes and the nodes (EC2’s) got stuck during their healthcheck in “1/2 checks passed” with the error - “Instance reachability check failed”.

attached our config :


I can try to run 0.36.0 tomorrow, where can I find the image?


```
driver:
  enabled: true
  kind: modern-bpf
```
Thanks for your help!

**Comment by Andreagit97** [comment]
thank you very much for testing it! Not sure we can provide this image for tomorrow but for sure we will provide you with a Falco dev image in these days!
The official Falco 0.36 release should be on September 25 but we will try to provide you with a dev image to test it before this date :) Thank you for helping in the debugging

**Comment by Andreagit97** [comment]
Ok @natalia-k you can use the falco-no-driver `master` image, so all you need to change in the helm chart is the `tag` of the falco image :point_down: 

```yaml
# Default values for Falco.

###############################
# General deployment settings #
###############################

image:
  # -- The image pull policy.
  pullPolicy: IfNotPresent
  # -- The image registry to pull from.
  registry: docker.io
  # -- The image repository to pull from
  repository: falcosecurity/falco-distroless
  # -- The image tag to pull. Overrides the image tag whose default is the chart appVersion.
  tag: "master" # -- <------- this one
```

**Comment by natalia-k** [comment]
@Andreagit97 I am running with tag: "master" now and it's ok for now, will update later after few hours
```
containerStatuses:
    - name: falco
      image: docker.io/falcosecurity/falco-no-driver:master
```
 but I have some misunderstanding, according the [doc](https://github.com/falcosecurity/charts/blob/master/falco/README.md#about-drivers): 
`The Modern eBPF probe doesn't require an init container because it is shipped directly into the Falco binary`
but I still see in pod's yaml "initContainer" :
```
driver:
  enabled: true
  kind: modern-bpf
  loader:
    initContainer:
      image:
        pullPolicy: IfNotPresent
        registry: ...
        repository: falcosecurity/falco-driver-loader
        tag: 0.35.1
```
does it ok?

and regarding : `Falco can automatically detect if these features are available on the running machine and can notify you if something is missing.` , if these features are unavailable should I see the "error" in falco logs immediately when pod starting ?

Thanks

**Comment by Andreagit97** [comment]
> The Modern eBPF probe doesn't require an init container because it is shipped directly into the Falco binary
but I still see in pod's yaml "initContainer" :

Hi, yes that's fine! The helm chart can automatically understand that with the modern-bpf you don't need the init-container, so if you check at startup time the falco-driver-loader won't be called

> and regarding : Falco can automatically detect if these features are available on the running machine and can notify you if something is missing. , if these features are unavailable should I see the "error" in falco logs immediately when pod starting ?

Yes exactly in the Falco log you should see something like:
```
tracing program type is not supported (errno: 22 | message: Invalid argument)
````

Let us know if you see again some crashes in the nodes :)

**Comment by natalia-k** [comment]
@Andreagit97 we are still running ok with tag: "master" !
do you know what was a root cause in v0.35.1 ? what is the change in v0.36 ? I need more details since we broke production with v0.35.1
Thanks a lot for the help!

**Comment by Andreagit97** [comment]
ei @natalia-k, yes the cause was probably a NULL dereference in bpf which could cause a kernel panic.
In Falco 0.36 we have these 2 fixes:
* https://github.com/falcosecurity/libs/pull/1236 
* https://github.com/falcosecurity/libs/pull/1251

to avoid possible null dereference. So now it should work! Sorry again for the production issue

**Comment by natalia-k** [comment]
@Andreagit97 Great! Thanks a lot for help, fix and explanation.
When v0.36 will be officially released ?

**Comment by Andreagit97** [comment]
> When v0.36 will be officially released ?

Next Monday, so the 25th of September :)

**Comment by FedeDP** [comment]
This should now be fixed, since Falco 0.36.0 was released 1 hr ago!

**Comment by Andreagit97** [comment]
I will close it since this should be solved by Falco 0.36.0, feel free to reopen it if this is not the case :)
